### PR TITLE
Bugfix: Datagrid search reload

### PR DESCRIPTION
**Pull request by trickreich**
| Q | A
| --- | ---
| Bug fix? | yes
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | fixes --
| Related issues/PRs | --
| License | MIT
| Documentation PR | --

#### What's in this PR?

Set value of datagrid search correctly when it's changing from above (for example router/store).

#### Why?

Current search wasn't visible in the input when opening direct link.

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/tests/Search.test.js [hunk 1]
```diff
@@ -11,15 +11,35 @@ jest.mock('../../../utils/Translator', () => ({
...
-        <Search onSearch={jest.fn()} />
+        <Search onSearch={jest.fn()} value={null} />
     );
 
     expect(search).toMatchSnapshot();
 });
 
+test('The component should render not collapsed when value is given', () => {
+    const search = shallow(
+        <Search onSearch={jest.fn()} value={'search-string'} />
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/tests/Search.test.js]: Until now we have written `value="search-string"`. Also valid for the other tests here.

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Search.js [hunk 2]
```diff
@@ -22,6 +23,24 @@ export default class Search extends React.Component<Props> {
         this.value = value;
     }
 
+    updateState(value: ?string) {
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Search.js]: `updateState` is a very generic name... Can't we come up with something more specific? Maybe `updateValue`?

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Datagrid.js [hunk 3]
```diff
@@ -132,7 +132,7 @@ export default class Datagrid extends React.Component<Props> {
             <Fragment>
                 <div className={datagridStyles.toolbar}>
                     {searchable &&
-                        <Search onSearch={this.handleSearch} />
+                        <Search onSearch={this.handleSearch} value={store.searchTerm ? store.searchTerm.get() : null} />
```
Comment (danrot) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Datagrid.js]: Is this check really necessary? From what I see `searchTerm` is always defined. Only the return value of `store.searchTerm.get()` could be undefined, but I think that would also be an acceptable option here.

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Search.js (hunk 2 above)
Comment (trickreich) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Datagrid/Search.js]: It's not just an update of the value here..

**Pull request merged**